### PR TITLE
Feature/deep update delete and detail-append

### PR DESCRIPTION
**Pull request by Jsyro**
special flag caller can add to json blob that tells the backend to delete nested object.

### CODE REVIEW

File: services/core-api/app/api/constants.py [hunk 1]
```diff
@@ -83,6 +83,9 @@ def DOWNLOAD_TOKEN(token_guid):
     return f'document-manager:download-token:{token_guid}'
 
 
+#Deep Update Special Flag
+DELETE_ITEM_FROM_LIST_JSON_KEY = "DELETE_ON_PUT"
```
Comment (nayr974) [inline services/core-api/app/api/constants.py]: After discussion let's make this a more generic state name that includes delete.

**Comment by pdittaro** [comment]
Jason and I had a discussion about behaviour differences between deleting records and disassociating records. E.g. remove a contact from a mine, but don't delete that contact. This will be something that we can expand into later with the 'state_modified' attribute.

Otherwise, this looks good.

File: services/core-api/app/api/utils/setup_marshmallow.py [hunk 2]
```diff
@@ -85,29 +85,29 @@ class ModelSchema2(ModelSchema):
                 except Exception as e:
                     raise e
 
-        for class_ in Base._decl_class_registry.values():
-            if hasattr(class_, "_schema"):
-                pprint(class_.__name__)
-                try:
-                    mapper = inspect(class_)
+        # for class_ in Base._decl_class_registry.values():
```
Comment (pdittaro) [inline services/core-api/app/api/utils/setup_marshmallow.py]: Are we going to be using this later?

Comment (pdittaro) [reply services/core-api/app/api/utils/setup_marshmallow.py]: Talked with Jason, these are fine to keep for now as this is WIP.

**Pull request merged**